valinta-tulos-service/src/main/scala/fi/vm/sade/valintatulosservice/migri/MigriService.scala

```diff
@@ -0,0 +1,99 @@
+                  if (hyvaksytytHakukohdeOidit.contains(hakukohde.oid)) {
+                    hakemusRepository.findHakemus(tulos.valinnantulos.hakemusOid).fold(e =>
+                      logger.warn(s"No hakemus found for migri hakijaOid: ${tulos.valinnantulos.hakemusOid}, cause: ${e.toString}"), h => {
+                      val lukuvuosimaksu: String = lukuvuosimaksuService.getLukuvuosimaksuByHakijaAndHakukohde(HakijaOid(h.henkiloOid), tulos.valinnantulos.hakukohdeOid, auditInfo) match {
+                        case Some(maksu) => maksu.maksuntila.toString
+                        case None => null
+                      }
+                      val maksuvelvollisuus: String = if (h.maksuvelvollisuudet.exists(m => m._1 == tulos.valinnantulos.hakukohdeOid.toString)) h.maksuvelvollisuudet.filter(m => m._1 == tulos.valinnantulos.hakukohdeOid.toString).head._2 else null
+                      val koulutuksenAlkamiskausi: String = hakukohde.koulutuksenAlkamiskausi match {
+                        case Some(kausi) => kausi
+                        case None => null
+                      }
+                      val koulutuksenAlkamisvuosi: Integer = hakukohde.koulutuksenAlkamisvuosi match {
+                        case Some(vuosi) => vuosi
+                        case None => null
+                      }
+
+                      hakija.hakemukset += MigriHakemus(
+                        hakuOid = h.hakuOid.toString,
+                        hakuNimi = hakukohde.hakuNimi,
+                        hakemusOid = h.oid.toString,
+                        organisaatioOid = hakukohde.organisaatioOid,
+                        organisaatioNimi = hakukohde.organisaatioNimi,
+                        hakukohdeOid = tulos.valinnantulos.hakukohdeOid.toString,
+                        hakukohdeNimi = hakukohde.hakukohteenNimi,
+                        toteutusOid = hakukohde.toteutusOid,
+                        toteutusNimi = hakukohde.toteutusNimi,
+                        valintaTila = tulos.valinnantulos.valinnantila.valinnantila.toString,
+                        vastaanottoTila = tulos.valinnantulos.vastaanottotila.toString,
+                        ilmoittautuminenTila = tulos.valinnantulos.ilmoittautumistila.ilmoittautumistila.toString,
+                        maksuvelvollisuus = maksuvelvollisuus,
+                        lukuvuosimaksu = lukuvuosimaksu,
+                        koulutuksenAlkamiskausi = koulutuksenAlkamiskausi,
+                        koulutuksenAlkamisvuosi = koulutuksenAlkamisvuosi
+                      )
+                    })
+                  }
+                case _ =>
+              }
+            }
+      }
+    })
+    foreignHakijat.filter(h => h.hakemukset.nonEmpty)
+  }
+
+  private def getHakukohdeMigri(hakukohdeOid: HakukohdeOid): Option[HakukohdeMigri] = {
+    try {
+      Some(hakuService.getHakukohdeMigri(hakukohdeOid).right.get)
+    } catch {
+      case e: Throwable =>
+        logger.warn(e.toString)
```

Comment:
Näin taitaa tosiaan käydä aina tarjontahakukohteiden tapauksessa, mikä on varmaan ok, mut onhan tämä kommunikoitu migrin suuntaan? Rajapinnan kutsujalle tää varmaan näyttäytyy niin että "hakemusta ei ole", vaikka oikeasti olisikin.

Reply:
Migrille tästä ollaan kerrottu ja vastaus on että voivat "elää asian kanssa" Jos vanhan tarjonnan aikaisia hakemuksia halutaan kaivella, täytyy se tehdä käsityönä.